# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/worklet/WorkletImpl.h
# Commit: a8e623b228ae
# Full Hash: a8e623b228aeb04c741bebeeaea4f2524b2ddcc7
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2022-11-29 21:47:49
# Regressor Bug: 1778510
# File Overlap Count: 1
# Description:
#   Bug 1778510: Add RTPCallerType to GlobalObject r=asuth
#   
#   This centralizes the logic in one place.
#   
#   In order to do this, we will need to check the principal
# ==============================================================================

diff -r 32fdf6f014f7 -r a8e623b228ae dom/worklet/WorkletImpl.h
--- a/dom/worklet/WorkletImpl.h	Tue Nov 29 13:34:14 2022 +0000
+++ b/dom/worklet/WorkletImpl.h	Tue Nov 29 13:34:15 2022 +0000
@@ -82,6 +82,7 @@
   const Maybe<nsID>& GetAgentClusterId() const { return mAgentClusterId; }
 
   bool IsSharedMemoryAllowed() const { return mSharedMemoryAllowed; }
+  bool IsSystemPrincipal() const { return mIsSystemPrincipal; }
   bool ShouldResistFingerprinting() const {
     return mShouldResistFingerprinting;
   }
@@ -106,6 +107,8 @@
   ipc::PrincipalInfo mPrincipalInfo;
   // Accessed on only worklet parent thread.
   nsCOMPtr<nsIPrincipal> mPrincipal;
+  // For off-main-thread checks
+  bool mIsSystemPrincipal;
 
   const WorkletLoadInfo mWorkletLoadInfo;
 
